src/wmtk/energy/ValenceEnergy.cpp

```diff
@@ -0,0 +1,54 @@
+#include "ValenceEnergy.hpp"
+#include <wmtk/SimplicialComplex.hpp>
+
+namespace wmtk {
+namespace energy {
+ValenceEnergy::ValenceEnergy(const TriMesh& mesh)
+    : Energy(mesh)
+{}
+
+double ValenceEnergy::energy_eval(const Tuple& tuple) const
+{
+    // assume tuple is not a boundary edge
+    const Tuple current_v = tuple;
+    const Tuple other_v = m_mesh.switch_vertex(tuple);
+    long val0 = static_cast<long>(SimplicialComplex::vertex_one_ring(mesh(), current_v).size());
```

Comment:
this valence definition is very specific to collapse. The ValenceEnergy should just return the valence - this calculation for predicting the before/after only makes sense as an Invariant